File: 2441-largest-positive-integer-that-exists-with-its-negative/2441-largest-positive-integer-that-exists-with-its-negative.java
class Solution {
    public int findMaxK(int[] nums) {
        Arrays.sort(nums);
        int i = 0;
        int j = nums.length - 1;
        
        while (i < j){
            if (nums[i] < 0) {
                int val = Math.abs(nums[i]);
                if (nums[j] == val) return nums[j];
                else if (nums[j] > val) j--;
                else if (nums[j] < val) i++;
            }
            else{
                return -1;
            }
        }
        return -1;
    }
}

File: 2441-largest-positive-integer-that-exists-with-its-negative/2441-largest-positive-integer-that-exists-with-its-negative.py
class Solution:
    def findMaxK(self, nums: List[int]) -> int:
        arr = sorted(nums)
        
        i, j = 0, len(arr)-1
        
        while i < j:
            if arr[i] < 0:
                if arr[j] == abs(arr[i]):
                    return arr[j]
                elif arr[j] > abs(arr[i]):
                    j -= 1
                elif arr[j] < abs(arr[i]):
                    i += 1
            else:
                return -1
                
        return -1